src/server/worker/datum/iterator_test.go

```diff
@@ -244,77 +244,40 @@ func TestIterators(t *testing.T) {
-		//	require.NoError(t, err)
-		//	validateDI(t, s3itr, "/")
-
-		//	// Check that every datum has an S3 input
-		//	s3itr, _ = NewIterator(c, in11)
-		//	var checked, s3Count int
-		//	for s3itr.Next() {
-		//		checked++
-		//		require.Equal(t, 1, len(s3itr.Datum()))
-		//		if s3itr.Datum()[0].S3 {
-		//			s3Count++
-		//			break
-		//		}
-		//	}
-		//	require.True(t, checked > 0 && checked == s3Count,
-		//		"checked: %v, s3Count: %v", checked, s3Count)
-		//})
-
-		//// in12 is a cross that contains an S3 input and two non-s3 inputs
-		//in12 := client.NewCrossInput(in1, in2, in11)
-		//t.Run("S3MixedCross", func(t *testing.T) {
-		//	s3CrossItr, err := NewIterator(c, in12)
-		//	require.NoError(t, err)
-		//	validateDI(t, s3CrossItr,
-		//		"/foo11/foo12/", "/foo21/foo12/", "/foo31/foo12/", "/foo41/foo12/",
-		//		"/foo11/foo2/", "/foo21/foo2/", "/foo31/foo2/", "/foo41/foo2/",
-		//		"/foo11/foo22/", "/foo21/foo22/", "/foo31/foo22/", "/foo41/foo22/",
-		//		"/foo11/foo32/", "/foo21/foo32/", "/foo31/foo32/", "/foo41/foo32/",
-		//		"/foo11/foo42/", "/foo21/foo42/", "/foo31/foo42/", "/foo41/foo42/",
-		//	)
-
-		//	s3CrossItr, _ = NewIterator(c, in12)
-		//	var checked, s3Count int
-		//	for s3CrossItr.Next() {
-		//		checked++
-		//		for _, d := range s3CrossItr.Datum() {
-		//			if d.S3 {
-		//				s3Count++
-		//			}
-		//		}
-		//	}
-		//	require.True(t, checked > 0 && checked == s3Count,
-		//		"checked: %v, s3Count: %v", checked, s3Count)
-		//})
+		// in27 is an S3 input
+		in27 := client.NewS3PFSInput("", dataRepo, "")
+		in27.Pfs.Commit = commit.ID
+		t.Run("PlainS3", func(t *testing.T) {
+			di, err := NewIterator(c, in27)
+			require.NoError(t, err)
+			validateDI(t, di, "/")
+			// Check that every datum has an S3 input
+			require.NoError(t, di.Iterate(func(meta *Meta) error {
+				require.True(t, meta.Inputs[0].S3)
+				return nil
+			}))
+		})
 
-		//// in13 is a cross consisting of exclusively S3 inputs
-		//in13 := client.NewCrossInput(in11, in11, in11)
-		//t.Run("S3OnlyCrossUnionJoin", func(t *testing.T) {
-		//	s3CrossItr, err := NewIterator(c, in13)
-		//	require.NoError(t, err)
-		//	validateDI(t, s3CrossItr, "///")
+		// in28 is a cross that contains an S3 input and two non-s3 inputs
+		in28 := client.NewCrossInput(in1, in2, in27)
+		t.Run("S3MixedCross", func(t *testing.T) {
+			di, err := NewIterator(c, in28)
+			require.NoError(t, err)
+			validateDI(t, di,
+				"/foo11/foo12/", "/foo11/foo2/", "/foo11/foo22/", "/foo11/foo32/", "/foo11/foo42/",
+				"/foo21/foo12/", "/foo21/foo2/", "/foo21/foo22/", "/foo21/foo32/", "/foo21/foo42/",
+				"/foo31/foo12/", "/foo31/foo2/", "/foo31/foo22/", "/foo31/foo32/", "/foo31/foo42/",
+				"/foo41/foo12/", "/foo41/foo2/", "/foo41/foo22/", "/foo41/foo32/", "/foo41/foo42/",
+			)
+		})
 
-		//	s3CrossItr, _ = NewIterator(c, in13)
-		//	var checked, s3Count int
-		//	for s3CrossItr.Next() {
-		//		checked++
-		//		for _, d := range s3CrossItr.Datum() {
-		//			if d.S3 {
```

Comment:
Yeah, I thought it was superfluous test code. I kind of think that all of these S3 input tests are overkill because it is just a PFS input with a field set, which is not influenced by the structure of the composition. I think the input composition tests just need to focus on correct composition, not whether a field in one of the composed inputs is set when it is spit out of the composition.